src/teleop.rs

```diff
@@ -0,0 +1,76 @@
+use crate::config::TeleopConfig;
+use rosrust;
+use rosrust_msg;
+use termion::event::Key;
+use std::collections::HashMap;
+
+pub struct Teleoperator{
+    current_velocities: Velocities,
+    _cmd_val_pub: rosrust::Publisher<rosrust_msg::geometry_msgs::Twist>,
+    _keys_to_direction: HashMap<Key, (String, i8)>,
+    _increment: f64,
+}
+
+pub struct Velocities{
+    x: f64,
+    y: f64,
+    theta: f64,
+}
+
+fn match_key_to_string(
+    string_to_direction: &HashMap<String, (String, i8)>,
+) -> HashMap<Key, (String, i8)>
+{
+    let mut keys_to_direction = HashMap::new();
+    for (string, val) in string_to_direction {
+        keys_to_direction.insert(Key::Char(string.chars().next().unwrap()), val.clone());
+    };
+    keys_to_direction
+}
+
+impl Teleoperator {
+    pub fn new(
+        config: TeleopConfig,
+    ) -> Teleoperator
+    {
+        let cmd_val_publisher = rosrust::publish("/cmd_val", 1).unwrap();
```

Comment:
fix typo: /cmd_vel

Reply:
already parametrizes, saw it a min too late